sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -115,7 +120,40 @@ public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON
         return reqBodyLoggingMono;
     }
 
-    private Function<HttpResponse, Mono<HttpResponse>> logResponseDelegate(final ClientLogger logger, final URL url,
+    private void formatAllowableHeaders(Set<String> allowedHeaderNames, HttpRequest request, ClientLogger logger) {
+        for (HttpHeader header : request.getHeaders()) {
+            if (allowedHeaderNames.contains(header)) {
+                logger.info(header.toString());
+            } else {
+                logger.info(REDACTED_PLACEHOLDER);
```

Comment:
For a query parameter pair we log key as it is and value as REDACTED i.e. 
i.e. `a=b` as `a=REDACTED`, should we do the same for header as well? e.g: `Authorization:REDACTED`?

Reply:
yup, added the same thing with Headers.